src/vs/vscode.proposed.d.ts

```diff
@@ -681,4 +681,19 @@ declare module 'vscode' {
 	}
 
 	//#endregion
+
+	//#region Matt: Deinition range
+
+	export interface DefinitionLink {
+		origin?: Range;
+		uri: Uri;
+		range: Range;
+		selectionRange?: Range;
```

Comment:
One a second thought I wonder if defaulting `DefinitionLink.range` to `Location.range` is the right thing... It feels like its the `selectionRange` cos that how it is currently used. It's also OK to leave this out of this PR but something we should likely do when we optimise how the preview hover is computed.